### ISSUE TITLE
Error at the CombineAssets

### ISSUE THREAD

**Issue by rizhenkov** [octobercms/october#3264]
##### Expected behavior
When I add .less file in the plugin component, I expext that file will be compiled to .css as described in docs and added to styles.

##### Actual behavior
When I add .less file:
```php
public function onRun(){
    $this->addCss(['assets/less/main.less']);
}
```
System tryed to load file from incorrect path started from root and fails with error. 

File (really) located here (absolute path): 
`/home/site/plugins/company/pluginname/assets/less/main.less`

Path from the error (where system trying to get it).
```
file_get_contents(/plugins/company/pluginname/assets/less/main.less): failed to open stream: No such file or directory
```

##### Reproduce steps

1. Create empty plugin and add component
2. Create plugin less file at the assets/less/main.less
3. Add less file to styles from component's onRun method
```php
public function onRun(){
    $this->addCss(['assets/less/main.less']);
}
```
4. Add component to page and try to access it from public. It fails.

##### October build
428

First candidate where is may be bug:
class CombineAssets at line 456  (I think, error in the unnamed function)
```php
protected function getDeepHashFromAssets($assets)
    {
        $key = '';

        $assetFiles = array_map(function ($file) {
            return file_exists($file) ? $file : File::symbolizePath($file, null) ?: $this->localPath . $file;
        }, $assets);

```

Stack trace: 
![default](https://user-images.githubusercontent.com/987517/33216984-32a84632-d147-11e7-93de-6537c62d71db.png)

**Comment by LukeTowers** [comment]
The AssetMaker trait sets `$this->assetPath` to a public path (relative to the project) by default instead of an absolute path (required for the asset combiner). Try this instead:
```php
public function onRun()
{
    $publicAssetPath = $this->assetPath;
    $this->assetPath = $this->guessViewPath('/assets');
    $this->addCss(['assets/less/main.less']);
    $this->assetPath = $publicAssetPath;
}
```

**Comment by LukeTowers** [comment]
But you're right, this behaviour could be improved, perhaps to try using an absoluteAssetPath when using the combiner. We would need to maintain backwards compatibility with any changes though.

**Comment by riartem** [comment]
Thank you for a quick solution. Works! 👍 
But I really think that was a bug and need for fix :)

**Comment by miy4mori** [comment]
same issue.
finally,I have to use css instead of scss.

**Comment by LukeTowers** [comment]
@miy4mori did you try the solution: https://github.com/octobercms/october/issues/3264#issuecomment-346857066?

**Comment by miy4mori** [comment]
@LukeTowers Yes, I tried, but in my case it just did not throw an exception, and scss does not seem to be compiled correctly. I did not carefully verify in a hurry , I will take the time to do more verification.

**Comment by LukeTowers** [comment]
@miy4mori Keep in mind that each array element is its own compilation context, i.e. `['variables.scss', 'main.scss']` won't work because `variables.scss` won't be included in `main.scss`. To get around that you `@include variables.scss` in `main.scss`. The other thing is that the current version of the SCSS compilation library may not support some of the features you are trying to use, look at the compiled output and check to see if there are any error messages in it.

**Comment by cleverer** [comment]
I had this problem as well. I think this definitely needs to be fixed, since the example of the documentation doesn't work: https://octobercms.com/docs/cms/pages#injecting-assets
Or am I missing something?

**Comment by LukeTowers** [comment]
@dinosauriel @cleverer please post exactly what you have tried in detail

**Comment by cleverer** [comment]
##### Expected behavior
Assets get included correctly.

##### Actual behavior
An error ist thrown.
```
The source "/themes/demo/assets/test.css" is not in the root directory "/Users/username/Sites/test"
/Users/username/Sites/test/vendor/kriswallsmith/assetic/src/Assetic/Asset/FileAsset.php line 46
```
##### Reproduce steps
1. Fresh install of October (use default settings with postgresql)
2. Add a new asset called `test.css`
3. Add a new page with the following content: `{% styles %}`.
4. Add the following code in the code section:  
```php
<?php
function onStart() {
    $this->addCSS(['assets/test.css']);
}
?>
```
5. Preview the page.

##### October build
436 on macOS 10.13.5, PHP-FPM 7.2.6

**Comment by LukeTowers** [comment]
@cleverer see https://github.com/octobercms/october/issues/3264#issuecomment-346857066. Additionally, if all you need to do is include a CSS file, then do that without the `[]` syntax that invokes the asset combiner.

**Comment by cleverer** [comment]
@LukeTowers I've seen the workaround. But I'm not trying to make it work, but rather propose, that it should work out of the box. Or, at least, it should be mentioned in the documentation. It's very unintuitive for a beginner, which tries to reproduce the examples in the documentation.

I am specifically using the asset combiner, since I want the css to get minified an maybe add additional style files later on. So the `[]` are required.

**Comment by LukeTowers** [comment]
@cleverer what changes to the documentation would you recommend? Shoot a PR over to octobercms/docs and I'll be happy to merge it. Alternatively, do you have any ideas for making it more intuitive by default?

**Comment by cleverer** [comment]
@LukeTowers No idea how to implement this, but I think adding a relative path should automatically be expanded before handed to the assetmaker. Or at least that's what I had expected originally.

If this is not possible/too complicated I suggest the solution in https://github.com/octobercms/october/issues/3264#issuecomment-346857066 gets added to the documentation at https://octobercms.com/docs/cms/pages#injecting-assets. I can try to do that…

**Comment by LukeTowers** [comment]
@cleverer sure, give adding it to the docs a shot! The best solution would be to separate the local filesystem asset path and the public URL asset path variables but I don't have time to do that right now so adding that to the docs for now would be a good temporary solution unless someone wants to implement a full fix for this.

**Comment by cleverer** [comment]
With 87ed09166d0b2284f159f0d0d34895a16731eb4c I'd like to reopen this Issue and propose an easy solution. Could someone give me feedback on my commit? If it looks good, I'd open a PR.

As an easier workaround than https://github.com/octobercms/october/issues/3264#issuecomment-346857066, one can just prepend the Path with `~/themes/foo/` for a theme called foo. So in my example above (https://github.com/octobercms/october/issues/3264#issuecomment-39874598) the complete code would become:
```php
<?php
function onStart() {
    $this->addCSS(['~/themes/demo/assets/test.css']);
}
?>
```

**Comment by cleverer** [comment]
Opened PR #3802 for easier discussion of code.

**Comment by LukeTowers** [comment]
Fixed by https://github.com/octobercms/october/commit/aeb7616d063121d47cfb5aefe07fa9524e4a522b
